pkg/services/sqlstore/permissions/dashboard.go

```diff
@@ -114,8 +114,21 @@ func (f *accessControlDashboardPermissionFilter) Where() (string, []any) {
 	return f.where.string, f.where.params
 }
 
+// Check if user has no permissions required for search to skip expensive query
+func (f *accessControlDashboardPermissionFilter) hasRequiredActions() bool {
```

Comment:
@IevaVasiljeva This is a more accurate way to check all required permissions. We could skip "shared with me" in this case, it's folder permission, so it's only required when `/api/search?type=dash-folder` search is using, but I checked out, it's not affected by slow query bug.